flink-libraries/flink-table/src/main/scala/org/apache/flink/table/plan/nodes/datastream/DataStreamOverAggregate.scala

```diff
@@ -106,9 +113,14 @@ class DataStreamOverAggregate(
         if (overWindow.lowerBound.isUnbounded &&
           overWindow.upperBound.isCurrentRow) {
           createUnboundedAndCurrentRowProcessingTimeOverWindow(inputDS)
+        } // lowerbound is a BasicType and upperbound is PRECEEDING or CURRENT ROW
+        else if (overWindow.lowerBound.getOffset.getType.isInstanceOf[BasicSqlType]
```

Comment:
Yes, I realized that when looking at PR #3550 that `isInstanceOf[BasicSqlType]` and `.isInstanceOf[IntervalSqlType]` distinguishes ROW from RANGE windows. I think using `.isRows()` is more clear and might also be safer because it appears to be a more public API than the type of the offset.